### PR TITLE
Improved error handling

### PR DESCRIPTION
**Pull request by neelSharma12**
## Summary
This PR contains refactored code for better error handling. 

## Tested scenarios
I have tested below components: 
1. AffirmComponent
2. BoletoComponent
3. DokuComponent
4. MBWayComponent
5. QiwiWalletComponent

**Fixed issue**: 
[COIOS-537](https://youtrack.is.adyen.com/issue/COIOS-537)

### CODE REVIEW

File: Tests/AdyenTests/Adyen Tests/Components/Affirm/AffirmComponentTests.swift [hunk 1]
```diff
@@ -58,7 +58,7 @@ class AffirmComponentTests: XCTestCase {
         sut.deliveryAddressToggleItem.value = false
         
         // When
-        let paymentDetails = sut.createPaymentDetails()
+        let paymentDetails = try sut.createPaymentDetails()
```
Comment (erenbesel) [inline Tests/AdyenTests/Adyen Tests/Components/Affirm/AffirmComponentTests.swift:61]: Now that this function can throw, let's test a few failed scenarios with missing data

File: AdyenComponents/Boleto/BoletoComponent.swift [hunk 2]
```diff
@@ -87,7 +87,15 @@ public final class BoletoComponent: PaymentComponent, LoadingComponent, Presenta
...
                                       configuration: configuration,
-                                      onCreatePaymentDetails: { [weak self] in self?.createPaymentDetails() })
+                                      onCreatePaymentDetails: { [weak self] in
+            var paymentMethodDetails: PaymentMethodDetails?
+            do {
+                paymentMethodDetails = try self?.createPaymentDetails()
+            } catch let error {
+                adyenPrint(error)
+            }
+            return paymentMethodDetails
```
Comment (descorp) [inline AdyenComponents/Boleto/BoletoComponent.swift:97]: So in case of error we are going to print message and then use [InstantPaymentDetails(type: paymentMethod.type)](https://github.com/Adyen/adyen-ios/pull/864/files#diff-1b2fc8c8f5c7035f9237a992a77bcb221aba88bf22a1896d8cf5a2bf871c7da1L232) that will (most likely) fail on `\payment`.

I guess we need to refactor Boleto component entirely (in another PR) 🤔

**Review by descorp — APPROVED** [review]

File: AdyenComponents/Boleto/BoletoComponent.swift [hunk 3]
```diff
@@ -87,7 +87,15 @@ public final class BoletoComponent: PaymentComponent, LoadingComponent, Presenta
                                       apiContext: apiContext,
                                       fields: formFields,
                                       configuration: configuration,
-                                      onCreatePaymentDetails: { [weak self] in self?.createPaymentDetails() })
+                                      onCreatePaymentDetails: { [weak self] in
+            var paymentMethodDetails: PaymentMethodDetails?
+            do {
+                paymentMethodDetails = try self?.createPaymentDetails()
+            } catch let error {
+                adyenPrint(error)
```
Comment (erenbesel) [inline AdyenComponents/Boleto/BoletoComponent.swift:95]: Hmm, would printing the error be enough? 🤔

Comment (descorp) [reply AdyenComponents/Boleto/BoletoComponent.swift:95]: Not really. See [my comment](https://github.com/Adyen/adyen-ios/pull/864#discussion_r854081354) above
In addition - I am not sure that we can even test it. Since this scenario is caused by "Optional handling", rather then an actual business edge case..

Comment (erenbesel) [reply AdyenComponents/Boleto/BoletoComponent.swift:95]: Yeah, seems we need some refactoring

Comment (mohammedDehairy) [reply AdyenComponents/Boleto/BoletoComponent.swift:95]: I thinK we can make the closure throw, right?

**Review by mohammedDehairy — APPROVED** [review]
